src/components/DevicesModaleConfigureView.jsx

```diff
@@ -0,0 +1,305 @@
+      disableRipple={true}
+      readOnly
+    />
+  </IconButton>
+)
+
+const FolderRow = ({
+  name,
+  partialSync,
+  setPartialSync,
+  isIncluded,
+  toggleInclusion
+}) => (
+  <ListItem
+    divider
+    button
+    onClick={() => {
+      setPartialSync(true)
+      toggleInclusion()
+    }}
+  >
+    <ListItemIcon>
+      <Icon icon={FileTypeFolderIcon} width="24" height="24" />
+    </ListItemIcon>
+    <ListItemText>
+      <Typography variant="body1">{name}</Typography>
+    </ListItemText>
+    <FolderCheckbox checked={isIncluded} disabled={!partialSync} />
+  </ListItem>
+)
+
+const FoldersList = ({
+  folders,
+  partialSync,
+  excluded,
+  setPartialSync,
+  isIncluded,
+  toggleInclusion,
+  toggleAllInclusion
+}) => {
+  const { t } = useI18n()
+
+  return (
+    <Card className="u-mt-1 u-p-0">
+      <CardHeader
+        title={
+          <Typography variant="subtitle1" className="u-coolGrey">
+            {t('configureDevice.myCozy')}
+          </Typography>
+        }
+        action={
+          <FolderCheckbox
+            checked={excluded.length !== folders.length}
+            mixed={excluded.length > 0 && excluded.length !== folders.length}
+            disabled={!partialSync}
+            onClick={() => {
+              setPartialSync(true)
+              toggleAllInclusion()
+            }}
+          />
+        }
+        classes={{
+          root: 'u-pv-half u-ph-1-half u-bg-paleGrey u-fz-tiny',
+          action: 'u-mv-0'
+        }}
+      />
+      <CardContent className="u-p-0">
+        <List dense>
+          {folders.map(folder => (
+            <FolderRow
+              key={folder.id}
+              name={folder.name}
+              partialSync={partialSync}
+              setPartialSync={setPartialSync}
+              isIncluded={isIncluded(folder)}
+              toggleInclusion={() => toggleInclusion(folder)}
+            />
+          ))}
+        </List>
+      </CardContent>
+    </Card>
+  )
+}
+
+const ConfigureDeviceSyncDialog = ({
+  device,
+  cancelAction,
+  onDeviceConfigured
+}) => {
+  const { t } = useI18n()
+  const client = useClient()
+
+  const [loaded, setLoaded] = useState(false)
+  const [wasLoading, setWasLoading] = useState(false)
+  const [partialSync, setPartialSync] = useState(false)
+  const [wasExcluded, setWasExcluded] = useState([])
+  const [excluded, setExcluded] = useState([])
+
+  const foldersQuery = buildFoldersQuery({ currentFolderId: ROOT_FOLDER_ID })
+  const queryResult = useQuery(foldersQuery.definition, foldersQuery.options)
+
+  useMemo(() => {
+    // We wait until the query has been reloaded (and thus data has been fetched
+    // again) before marking our componant as loaded to avoid glitches between
+    // stale data loaded from the store and new data fetched from the Cozy.
+    if (!wasLoading && isQueryLoading(queryResult)) {
+      setWasLoading(true)
+    } else if (wasLoading && !loaded && !isQueryLoading(queryResult)) {
```

Comment:
if the request fails the first time, we'll pass in that block, right? I think this can be avoided by using `hasQueryBeenLoaded` instead?

Reply:
`hasQueryBeenLoaded` does something entirely different as it only checks if the query result has a `lastFetch` attribute value.
But I can re-order the conditions because you're right, we'd go through this block if the query failed.

Reply:
(Done. The failed query conditional block now comes before this one.)